.github/workflows/ci-build.yml

```diff
@@ -9,17 +9,46 @@ on:
 
 jobs:
   push_to_registry:
-    name: Push Docker image to Docker Hub
+    name: Build and Push Docker image to Docker Hub
     runs-on: ubuntu-18.04
 
     steps:
-      - name: Check out the repo
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]${GITHUB_REF#refs/heads/}"
+        id: extract_branch
+
+      - name: Check out the ${{ github.event.inputs.tag }} tag to build
         uses: actions/checkout@v2
+        with:
+          ref: refs/tags/${{ github.event.inputs.tag }}
+
+      - name: Check out the ${{ steps.extract_branch.outputs.branch }} branch for Dockerfile
+        uses: actions/checkout@v2
+        with:
+          ref: ${{github.ref}}
+          path: main_branch
 
-      - name: Push to Docker Hub
-        uses: docker/build-push-action@v1
+      - name: Copy Dockerfile from ${{ steps.extract_branch.outputs.branch }} branch to ${{ github.event.inputs.tag }}
+        run: |
+          cp ${{ github. workspace }}/main_branch/Dockerfile Dockerfile
+          rm -rf ${{ github. workspace }}/main_branch
+
+      - name: Set up QEMU
```

Comment:
What is this for? Qemu and buildx?

Reply:
The new version of build-push-action@v2 uses buildx and qemu to build docker images for multiple architectures.Hence its included